### ISSUE TITLE
Clippy does not respect inline #[allow(…)] attribute

### ISSUE THREAD

**Issue by selfhoster1312** [rust-lang/rust-clippy#17664]
### Summary

Very similar to #2332, clippy warning/suggestion remain after adding `allow`

### Lint Name

needless_pass_by_value

### Reproducer

I tried this code:

```rust
    #[allow(clippy::needless_pass_by_value)]
    sr: LdapSearchRequest,
```

I saw this happen:

```
error: this argument is passed by value, but not consumed in the function body
  --> src/op/search.rs:94:9
   |
94 |     sr: LdapSearchRequest,
   |         ^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/main/index.html#needless_pass_by_value
   = note: `-D clippy::needless-pass-by-value` implied by `-D clippy::pedantic`
   = help: to override `-D clippy::pedantic` add `#[allow(clippy::needless_pass_by_value)]`
help: consider taking a reference instead
   |
94 |     sr: &LdapSearchRequest,
   |         +
```

I expected to see this happen: nothing

### Version

```text
rustc 1.100.0-nightly (908501772 2026-08-30)
binary: rustc
commit-hash: 90850177249efe0321573c569aec5d12b257f8d6
commit-date: 2026-08-30
host: x86_64-unknown-linux-gnu
release: 1.100.0-nightly
LLVM version: 23.1.0
```

**Comment by Jarcho** [comment]
The lint will need to be changed to use `span_lint_hir` with the parameters `HirId`.

**Comment by selfhoster1312** [comment]
Thanks for replying quickly. Initial testing does not seem to be working:

```diff
diff --git a/clippy_lints/src/needless_pass_by_value.rs b/clippy_lints/src/needless_pass_by_value.rs
index c0b2298..f8530a9 100644
--- a/clippy_lints/src/needless_pass_by_value.rs
+++ b/clippy_lints/src/needless_pass_by_value.rs
@@ -1,4 +1,4 @@
-use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::diagnostics::span_lint_hir_and_then;
 use clippy_utils::res::{MaybeDef as _, MaybeResPath as _};
 use clippy_utils::source::{SpanExt as _, snippet};
 use clippy_utils::ty::{implements_trait, implements_trait_with_env_from_iter, is_copy};
@@ -295,9 +295,10 @@ impl<'tcx> LateLintPass<'tcx> for NeedlessPassByValue {
                     );
                 };

-                span_lint_and_then(
+                span_lint_hir_and_then(
                     cx,
                     NEEDLESS_PASS_BY_VALUE,
+                    hir_id,
                     input.span,
                     "this argument is passed by value, but not consumed in the function body",
                     sugg,
```

Running `cargo dev setup toolchain` (as per [clippy docs](https://doc.rust-lang.org/clippy/development/basics.html)) and `cargo +clippy clippy` in my project still produces the error.

To be honest, i have no idea what i'm looking at when i look at this lint code. I gave it a short try, but i'm happy to let someone else tkae the lead on this. (i will of course read the proposed PR and probably learn some things along the way!)

**Comment by Jarcho** [comment]
That's the function's `HirId`. You want `arg.hir_id` there.

**Comment by selfhoster1312** [comment]
> That's the function's `HirId`. You want `arg.hir_id` there.

I can confirm that's the correct way to fix the problem, thank you. Somebody beat me to opening the PR, which is great because i also had no idea how to add a test case for this 👍
